### PR TITLE
Agent runtime & Agent definitions from docker image labels.

### PR DESCRIPTION
**Pull request by deadly-panda**

### CODE REVIEW

File: src/ostorlab/runtimes/local/agent_runtime.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+from ostorlab.runtimes.local import runtime
+from ostorlab.runtimes import definitions
+from ostorlab.agent import definitions as agent_definitions
+from ostorlab.runtimes.local.services import mq
+
+
+
+class AgentRuntime:
+    """Methods to consolidate the agent settings and agent default defintion, and create the agent service."""
```
Comment (3asm) [inline src/ostorlab/runtimes/local/agent_runtime.py:20]:
```suggestion
    """Clas to consolidate the agent settings and agent default definition, and create the agent service."""
```

File: src/ostorlab/runtimes/local/agent_runtime.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+    def __init__(self,
+                 agent: definitions.AgentSettings,
```
Comment (3asm) [inline src/ostorlab/runtimes/local/agent_runtime.py:23]:
```suggestion
                 agent_settings: definitions.AgentSettings,
```

File: src/ostorlab/runtimes/local/agent_runtime.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            docker_client: docker client.
+        """
+        self._docker_client = docker_client
+        self.agent = agent
+        self.image_name = agent.container_image.split(':', maxsplit=1)[0]
+        self.runtime_name = runtime_name
+        self.mq_service = mq_service
+        self.update_agent_settings()
+
+    def created_settings_config(self) -> docker.types.ConfigReference:
```
Comment (3asm) [inline src/ostorlab/runtimes/local/agent_runtime.py:41]:
```suggestion
    def create_settings_config(self) -> docker.types.ConfigReference:
```

File: src/ostorlab/runtimes/local/agent_runtime.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        agent_instance_settings_proto = self.agent.to_raw_proto()
+        config_name = f'config_settings_{self.image_name}_{self.runtime_name}'
+        docker_config = self._docker_client.configs.create(name=config_name,
+                                                           labels={'ostorlab.universe': self.runtime_name},
+                                                           data=agent_instance_settings_proto)
+        return docker.types.ConfigReference(config_id=docker_config.id,
+                                            config_name=config_name,
+                                            filename='/tmp/settings.binproto')
+
+    def created_definition_config(self) -> docker.types.ConfigReference:
```
Comment (3asm) [inline src/ostorlab/runtimes/local/agent_runtime.py:56]:
```suggestion
    def create_definition_config(self) -> docker.types.ConfigReference:
```

File: src/ostorlab/runtimes/local/agent_runtime.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        agent_definition = self._docker_client.images.get(self.agent.container_image).labels.get('agent_definition')
+        config_name = f'config_definition_{self.image_name}__{self.runtime_name}'
+        docker_config = self._docker_client.configs.create(name=config_name,
+                                                           labels={'ostorlab.universe': self.runtime_name},
+                                                           data=str.encode(agent_definition))
+        return docker.types.ConfigReference(config_id=docker_config.id,
+                                            config_name=config_name,
+                                            filename='/tmp/ostorlab.yaml')
+
+    def agent_definition_from_label(self) -> agent_definitions.AgentDefinition:
```
Comment (3asm) [inline src/ostorlab/runtimes/local/agent_runtime.py:71]:
```suggestion
    def create_agent_definition_from_label(self) -> agent_definitions.AgentDefinition:
```

**Review by 3asm — CHANGES_REQUESTED** [review]

File: src/ostorlab/runtimes/local/agent_runtime.py (hunk 2 above)
Comment (deadly-panda) [reply src/ostorlab/runtimes/local/agent_runtime.py:23]: what about the attribute itself ? i leave it as agent, or update it too ?

**Review by 3asm — APPROVED** [review]

**Pull request merged**